contracts/modules/cover/Cover.sol

```diff
@@ -619,6 +653,23 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     return (1 << coverAsset) & assetsBitMap > 0;
   }
 
+  function isPoolAllowed(uint productId, uint poolId) external view returns (bool) {
+
+    uint poolCount = allowedPools[productId].length;
+
+    if (poolCount == 0) {
+      return true;
```

Comment:
i see the intention here but doesn't match what i know to be the spec.

If useFixedPricing, you should check what pools are enabled.

If no pool is registered no pool is allowed.

If you interpret empty list as all allowed, we might as well remove the boolean

Reply:
We've agreed to implement fixed pricing and pool allowlist as separate features. It doesn't make sense to have a product which is not allowed on any pool - if we want to disable it we have the deprecated flag for that.